File: basic_python/practices/Sorting/Auxilary_Space_O(1)/Swap Sort/swap_sort.cpp
#include <iostream>
#include <vector>
#include <iomanip>
#include <chrono>

using namespace std;

/*
    Swap Sort (O(n^2))
    Compare every pair (i, j) with j > i and swap if out of order
*/
vector<double> using_swap(vector<double> arr) {
    int n = arr.size();
    for (int i = 0; i < n; ++i) {
        for (int j = i + 1; j < n; ++j) {
            if (arr[i] > arr[j]) {
                swap(arr[i], arr[j]);
            }
        }
    }
    return arr;
}

int main() {
    int n;
    cout << "input length of your array: ";
    cin >> n;

    vector<double> arr(n);
    for (int i = 0; i < n; ++i) {
        cout << "\t element " << i + 1 << ": ";
        cin >> arr[i];
    }

    cout << "\nYour input:\n";
    for (double x : arr) cout << x << " ";
    cout << "\n";

    cout << "Swap-Sort results:\n";

    // Start timer
    auto t0 = chrono::high_resolution_clock::now();

    vector<double> res = using_swap(arr);

    // End timer
    auto t1 = chrono::high_resolution_clock::now();
    chrono::duration<double> elapsed = t1 - t0;

    for (double x : res) cout << x << " ";
    cout << "\n";

    cout << fixed << setprecision(6);
    cout << "Elapsed time: " << elapsed.count() << " seconds\n";

    return 0;
}

File: basic_python/practices/Sorting/Auxilary_Space_O(1)/Swap Sort/swap_sort.py
import time

def using_swap(arr: list[float]) -> list[float]:
    n = len(arr)
    for i in range(n):
        for j in range(i + 1, n):
            if arr[i] > arr[j]:
                arr[i], arr[j] = arr[j], arr[i]
    return arr

if __name__ == "__main__":
    n = int( input("input length of your array:") )
    arr = []
    for _ in range(n):
        arr.append( float( input(f"\t element {_ + 1}: ") ) )
    print("Your input:")
    print(arr)
    print('Swap-Sort results:')
    t0 = time.time()
    res = using_swap(arr)
    print(res)
    print(f"Ellapsed time: {(time.time() - t0)}")